### PR TITLE
[Merged by Bors] - feat: Port algebra.order.EuclideanAbsoluteValue

### PR DESCRIPTION
**Pull request by zaxioms0**
Small PR. Nothing super special here, although I did have a question. On line 74, is it within style guidelines to use `@` to make the parameters explicit? Thanks!

### CODE REVIEW

**Comment by dupuisf** [comment]
> On line 74, is it within style guidelines to use `@` to make the parameters explicit? Thanks!

The best way to ask a question like this is to start a review of your own PR in which you add the comment directly on the relevant line. And if the issue gets resolved (which seems to be the case here), you can clearly mark it as such.

File: Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Order.AbsoluteValue
+import Mathlib.Algebra.EuclideanDomain.Instances
+
+/-!
+# Euclidean absolute values
+
+This file defines a predicate `absolute_value.is_euclidean abv` stating the
```
Comment (dupuisf) [inline Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean:17]:
```suggestion
This file defines a predicate `AbsoluteValue.IsEuclidean abv` stating the
```

File: Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+absolute value is compatible with the Euclidean domain structure on its domain.
+
+## Main definitions
+
+ * `absolute_value.is_euclidean abv` is a predicate on absolute values on `R` mapping to `S`
+    that preserve the order on `R` arising from the Euclidean domain structure.
+ * `absolute_value.abs_is_euclidean` shows the "standard" absolute value on `ℤ`,
+   mapping negative `x` to `-x`, is euclidean.
```
Comment (dupuisf) [inline Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean:25]:
```suggestion
 * `AbsoluteValue.IsEuclidean abv` is a predicate on absolute values on `R` mapping to `S`
    that preserve the order on `R` arising from the Euclidean domain structure.
 * `AbsoluteValue.abs_isEuclidean` shows the "standard" absolute value on `ℤ`,
   mapping negative `x` to `-x`, is euclidean.
```

**Comment by xoers** [comment]
Ah ok. Good to know! Thanks!

File: Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+-/
+
+
+-- mathport name: «expr ≺ »
```
Comment (dupuisf) [inline Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean:29]:
```suggestion
```

File: Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+section OrderedSemiring
+
+variable {R S : Type _} [EuclideanDomain R] [OrderedSemiring S]
+
+variable (abv : AbsoluteValue R S)
+
+/-- An absolute value `abv : R → S` is Euclidean if it is compatible with the
+`euclidean_domain` structure on `R`, namely `abv` is strictly monotone with respect to the well
+founded relation `≺` on `R`. -/
```
Comment (dupuisf) [inline Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean:43]:
```suggestion
`EuclideanDomain` structure on `R`, namely `abv` is strictly monotone with respect to the well
founded relation `≺` on `R`. -/
```

File: Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+section Int
+
+open Int
+
+-- TODO: generalize to `linear_ordered_euclidean_domain`s if we ever get a definition of those
+/-- `abs : ℤ → ℤ` is a Euclidean absolute value -/
+protected theorem abs_is_euclidean : IsEuclidean (AbsoluteValue.abs : AbsoluteValue ℤ ℤ) :=
+  {
+    map_lt_map_iff' := fun {x y} =>
+      show abs x < abs y ↔ natAbs x < natAbs y by rw [abs_eq_natAbs, abs_eq_natAbs, ofNat_lt] }
```
Comment (dupuisf) [inline Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean:78]:
```suggestion
protected theorem abs_isEuclidean : IsEuclidean (AbsoluteValue.abs : AbsoluteValue ℤ ℤ) :=
  {  map_lt_map_iff' := fun {x y} =>
       show abs x < abs y ↔ natAbs x < natAbs y by rw [abs_eq_natAbs, abs_eq_natAbs, ofNat_lt] }
```

**Comment by dupuisf** [comment]
You can have a look at the new naming convention [here](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki) in the section "Aligning names". Since it's a bit more fine-grained than in mathlib3, mathport gets it wrong quite often and a lot of it has to be done manually.

**Comment by xoers** [comment]
Thanks! Sorry I'm new to porting and completely forgot to update the names in the docstrings. thank you so much!

File: Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+end IsEuclidean
+
+end OrderedSemiring
+
+section Int
+
+open Int
+
+-- TODO: generalize to `linear_ordered_euclidean_domain`s if we ever get a definition of those
```
Comment (ChrisHughes24) [inline Mathlib/Algebra/Order/EuclideanAbsoluteValue.lean:72]:
```suggestion
-- TODO: generalize to `LinearOrderedEuclideanDomain`s if we ever get a definition of those
```

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**